File: External-Tutorial-Dagger-Android-Sample/src/main/java/com/gabrielsamojlo/dagger_212/di/ActivityContributorModule.java
package com.gabrielsamojlo.dagger_212.di;

import com.gabrielsamojlo.dagger_212.MainActivity;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class ActivityContributorModule {

    @ContributesAndroidInjector
    abstract MainActivity contributeMainActivity();

}

File: Tutorial6-4ComponentFactoryDagger-Android/src/main/java/com/smarttoolfactory/tutorial6_4componentfactorydagger_android/di/ActivityContributorModule.kt
package com.smarttoolfactory.tutorial6_4componentfactorydagger_android.di

import com.smarttoolfactory.tutorial6_4componentfactorydagger_android.MainActivity
import com.smarttoolfactory.tutorial6_4componentfactorydagger_android.di.module.MainActivityModule
import com.smarttoolfactory.tutorial6_4componentfactorydagger_android.di.scope.ActivityScope
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class ActivityContributorModule {

    @ActivityScope
    @ContributesAndroidInjector(modules = [MainActivityModule::class, FragmentContributorModule::class])
    abstract fun contributeMainActivity(): MainActivity

}